packages/assets-controllers/src/AccountTrackerController.ts

```diff
@@ -146,20 +195,46 @@ export class AccountTrackerController extends BaseControllerV1<
    * @param provider - Provider used to create a new underlying EthQuery instance.
    */
   set provider(provider: Provider) {
-    this.ethQuery = new EthQuery(provider);
+    this._provider = provider;
   }
 
   get provider() {
     throw new Error('Property only used for setting');
   }
 
+  /**
+   * Resolves a networkClientId to a network client config
+   * or globally selected network config if not provided
+   *
+   * @param networkClientId - Optional networkClientId to fetch a network client with
+   * @returns network client config
+   */
+  #getCorrectNetworkClient(networkClientId?: NetworkClientId): {
+    chainId: string;
+    ethQuery?: EthQuery;
+  } {
+    if (networkClientId) {
+      const networkClient = this.getNetworkClientById(networkClientId);
+
+      return {
+        chainId: networkClient.configuration.chainId,
+        ethQuery: new EthQuery(networkClient.provider),
+      };
+    }
+
+    return {
+      chainId: this.getCurrentChainId(),
+      ethQuery: this._provider ? new EthQuery(this._provider) : undefined,
+    };
+  }
+
   /**
    * Starts a new polling interval.
    *
    * @param interval - Polling interval trigger a 'refresh'.
    */
   async poll(interval?: number): Promise<void> {
-    const releaseLock = await this.mutex.acquire();
+    const releaseLock = await this.pollMutex.acquire();
```

Comment:
Just a note: I don't think it matters for performance but we're doubling up the mutex for the legacy polling.

Reply:
Actually is this necessary?

Reply:
I think this may be redundant since the refresh mutex is protecting against the only state updates we wouldn't want to happen concurrently?

Reply:
Removed https://github.com/MetaMask/core/pull/3586/commits/8982ab30bcb4d2bffaf1f01cf63b7c985efd1722